src/executable-code/executable-fragment.js

```diff
@@ -314,11 +314,25 @@ export default class ExecutableFragment extends ExecutableCodeTemplate {
       ? document.body
       : this.nodes[0].querySelector(SELECTORS.CANVAS_PLACEHOLDER_OUTPUT);
   }
-    
+
+  removeImports (completeSnippet, importSnippet) {
+    let newArr = [];
+
+    completeSnippet.map(function(val) {
+      importSnippet.includes(val) ? '' : newArr.push(val);
+    });
+
+    importSnippet.map(function(val) {
```

Comment:
Also, while this approach could work, what if we try to make it simpler?  We can keep the function signature, then if we expect to receive a couple of arrays, one being the complete snippet, and the other one being the import lines, this could be as easy as filtering the content of the `completeSnippet`, isn't it?

You can have a look to:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
Ok, that is also true.